### PR TITLE
restructured code to use APIManager singleton class. 

### PR DESCRIPTION
**Pull request by HariniSundaram0**
Now, I am able to display the song title of a song chosen right after authorization (hard coded to query for "Call Me Maybe" for now.

### CODE REVIEW

File: Project-Hotpot/Project-Hotpot/APIManager.swift [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        }
+
+        // MARK: - Accessors
+        
+        class func shared() -> APIManager {
+            return sharedAPIManager
+        }
+
+    
+    var lastPlayerState: SPTAppRemotePlayerState?
```
Comment (harshsinha) [inline Project-Hotpot/Project-Hotpot/APIManager.swift:37]: Dont know if this is iOS standard or not (@toby1236 can better comment) but generally it is good to define all vars at the top of the class followed by functions. Within those publics go before privates. Additionally statics go to the top

File: Project-Hotpot/Project-Hotpot/APIManager.swift [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    // MARK: - Spotify Authorization & Configuration
+    var responseCode: String? {
+        didSet {
+            fetchAccessToken { (dictionary, error) in
+                if let error = error {
+                    print("Fetching token request error \(error)")
```
Comment (harshsinha) [inline Project-Hotpot/Project-Hotpot/APIManager.swift:43]: Better to consistently use NSLog?

File: Project-Hotpot/Project-Hotpot/APIManager.swift [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+//  APIManager.swift
+//  Project-Hotpot
+//
+//  Created by Harini Sundaram on 7/11/22.
+//
+
+import Foundation
+import UIKit
+
+class APIManager: NSObject {
```
Comment (harshsinha) [inline Project-Hotpot/Project-Hotpot/APIManager.swift:11]: Nit: Consider having more explicit naming like SpotifyManager. API sounds generic and could mean this class handles all kinds of APIs like soundcloud etc.

**Review by harshsinha — COMMENTED** [review]
Good job creating the singleton manager class and modularizing your business logic away from your view code !

File: Project-Hotpot/Project-Hotpot/APIManager.swift (hunk 1 above)
Comment (toby1236) [reply Project-Hotpot/Project-Hotpot/APIManager.swift:37]: Ah yes, this is standard for iOS too. The more organized you keep mutable state, the easier it is to reason about bugs.

File: Project-Hotpot/Project-Hotpot/APIManager.swift (hunk 2 above)
Comment (toby1236) [reply Project-Hotpot/Project-Hotpot/APIManager.swift:43]: Yes, I'd say stick with one. Here's an interesting article comparing the two: https://stevenpcurtis.medium.com/logging-in-swift-d9b59146ff00

Either one is perfectly fine in my opinion, just a matter of how much additional detail you like in your logging method.

**Pull request merged**